app/src/main/java/ru/netology/nmedia/work/RefreshPosts.kt

```diff
@@ -32,4 +31,35 @@ class RefreshPostsWorker(
             Result.retry()
         }
     }
-}
\ No newline at end of file
+}
+
+class RefreshPostsWorkerFactory (
+    private val repository: PostRepository,
+) : WorkerFactory() {
+    override fun createWorker(
+        appContext: Context,
+        workerClassName: String,
+        workerParameters: WorkerParameters
+    ): ListenableWorker? = when (workerClassName) {
+        RefreshPostsWorker::class.java.name ->
+            RefreshPostsWorker(appContext, workerParameters, repository)
+        else ->
+            null
+    }
+}
```

Comment:
Занесите лучше фабрику во внутрь класса - у них так гарантированно никогда не разойдутся названия и лучше выражается намерени "для чего эта фабрика".
```suggestion
    class Factory (
        private val repository: PostRepository,
    ) : WorkerFactory() {
        override fun createWorker(
            appContext: Context,
            workerClassName: String,
            workerParameters: WorkerParameters
        ): ListenableWorker? = when (workerClassName) {
            RefreshPostsWorker::class.java.name ->
                RefreshPostsWorker(appContext, workerParameters, repository)
            else -> null
        }
    }
}
```